shiny/ui/_include_helpers.py

```diff
@@ -178,6 +184,73 @@ def include_css(
     return tags.link(dep, href=src, rel="stylesheet")
 
 
+@add_example()
+def include_bootstrap_css(
+    theme: Path | str | HTMLDependency,
+    bs_version: Optional[Version | str] = None,
+) -> list[HTMLDependency]:
+    """
+    Replace the built-in Shiny Bootstrap CSS theme.
+
+    Parameters
+    ----------
+    theme
+        The theme to include. It can be a path to a file, or an
+        :class:`~htmltools.HTMLDependency`.
+    bs_version
+        The version of Bootstrap used by this `theme`, used to check for compatibility
+        with the version of Bootstrap used by Shiny. If not provided, the version is
+        discovered from one of several places:
+
+        * If the :class:`~htmltools.HTMLDependency` includes a `bs_version` attribute,
+          that is used.
+        * If `theme` is a file, the version is discovered by parsing the Bootstrap
+          banner comment in the file, which should look something like
+          `* Bootstrap v4.5.2`.
+        * If `theme` is an :class:`~htmltools.HTMLDependency`, the first stylesheet in
+          the dependency is parsed for the version comment mentioned above.
+
+    Returns
+    -------
+    :
+        A list of HTMLDependencies, the specified `theme` and a special
+        :class:`~htmltools.HTMLDependency` to suppress Shiny's built-in Bootstrap CSS.
+
+    Raises
+    ------
+    RuntimeError
+        When the major versions of theme's `bs_version` and Shiny's built-in Bootstrap
+        version are different, indicating incompatibility between the theme's Bootstrap
+        version and the base version.
+
+    Warns
+    -----
+    RuntimeWarning
+        When the theme's Bootstrap version and Shiny's built-in Bootstrap version are
+        not identical but the major versions match, indicating a potential for
+        unexpected issues.
+
+    See Also
+    --------
+    * Use :func:`~shiny.ui.include_css` if your CSS file contains styles that should be
+      applied *in addition to* (rather than replacing) Shiny's built-in Bootstrap theme.
+
+    """
+    if isinstance(theme, (Path, str)):
+        theme_dep = head_content(include_css(theme))
+    else:
+        theme_dep = theme
+
+    # Check compatibility _after_ resolving the dependency so that
+    # `include_css()` has a chance to check that the theme file exists.
+    check_bootstrap_compatibility(theme, bs_version)
+
+    return [
+        *bootstrap_deps_suppress(["css"]),
+        theme_dep,
+    ]
```

Comment:
I know realize that this'll be problematic for multiple calls to `include_bootstrap_css()`. Maybe we could work around that by always returning the full Bootstrap dependency, but with the version incremented by `f".counter()"` each time?